server/src/controllers/leaderboardController.js

```diff
@@ -0,0 +1,27 @@
+import Leaderboard from '../database/models/leaderboard';
+
+const addScore = async (req, res) => {
+  const {
+    user: { username },
+    body: { score },
+  } = req;
+  const newLeaderboard = await Leaderboard.findOneAndUpdate(
+    { username },
+    { score },
```

Comment:
shouldn't we check if this is the highest score and only add if it is? Otherwise, a lower score might override previous high score.

Reply:
Yeah, nice catch. I will make this change.